org-code-javabuilder/playground/src/main/java/org/code/playground/Board.java

```diff
@@ -0,0 +1,120 @@
+     * 
+     * @param item the item to add. If the item is already on the board,
+     *             this method does nothing.
+     */
+    public void addClickableItem(ClickableItem item);
+
+    /**
+     * Removes the clickable item from the board.
+     * 
+     * @param item the item to remove. If the item is not on the board, this
+     *             method does nothing.
+     */
+    public void removeClickableItem(ClickableItem item);
+
+    /**
+     * Adds a non-clickable item to the board. 
+     *
+     * @param item the item to add. If the item is already on the board,
+     *             this method does nothing.
+     */
+    public void addItem(Item item, int x, int y, int width, int height);
+
+    /**
+     * Removes the non-clickable item from the board.
+     * 
+     * @param item the item to remove. If the image is not on the board, this
+     *             method does nothing.
+     */
+    public void removeItem(Item item);
+    
+    /**
+     * Draws text on the board.
+     *
+     * @param text the text to draw
+     * @param x the distance from the left side of the board to draw the text.
+     * @param y the distance from the top of the board to draw the text.
+     * @param color the color to draw the text.
+     * @param font the name of the font to draw the text in.
+     * @param fontStyle the name of the font style to draw the text in.
+     * @param height the height of the text in pixels.
+     * @param rotation the rotation or tilt of the text, in degrees
+     */
+    public drawText(String text, int x, int y, Color color, Font font, FontStyle fontStyle, int height, double rotation);
+
+    /**
+     * Draws text on the board with a normal font style
+     *
+     * @param text the text to draw
+     * @param x the distance from the left side of the board to draw the text.
+     * @param y the distance from the top of the board to draw the text.
+     * @param color the color to draw the text.
+     * @param font the name of the font to draw the text in.
+     * @param height the height of the text in pixels.
+     * @param rotation the rotation or tilt of the text, in degrees.
+     */
+    public drawText(String text, int x, int y, Color color, Font font, int height, double rotation);
+
+    /**
+     * Starts the playground game, waiting for the user to click on images and
+     * executing the appropriate code. To end the game, call the end() method. The
+     * run() method may only be called once per execution of a program.
+     * 
+     * @throws PlaygroundRunningException if the run() method has already been
+     *                                    called.
+     */
+    public void run() throws PlaygroundRunningException;
+
+    /**
+     * Ends the game, plays the sound supplied, and stops program execution.
+     * @param endingSound the name of a sound file in the asset manager to play at the end of the game.
+     * @throws PlaygroundNotRunningException if the run() method has not been
```

Comment:
should this also be thrown if exit is called twice or is that ok?

Reply:
I think it's fine to be a no-op if they somehow manage to call this twice.